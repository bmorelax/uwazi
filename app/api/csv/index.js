import csvLoader from './csvLoader.js';
import CSVExporter from './csvExporter';

export default csvLoader;
export { CSVExporter };
