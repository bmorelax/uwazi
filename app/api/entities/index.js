/** @format */

import model, { MetadataObject } from './entitiesModel';
import * as endpointSchema from './endpointSchema';

export { default } from './entities.js';
export { model, MetadataObject, endpointSchema };
