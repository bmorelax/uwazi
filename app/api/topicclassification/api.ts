/** @format */

import 'isomorphic-fetch';
import isReachable from 'is-reachable';
import { tcServer } from 'api/config/topicclassification';
import { URL } from 'url';

export async function getModels() {
  const tcUrl = new URL('models', tcServer);
  const msTimeout = 1000;
  if (!(await isReachable(tcUrl.href, { timeout: msTimeout }))) {
    // TODO: move this backend check to server start-up time, maybe
    return {
      models: {},
      error: `Topic Classification server is unreachable (${msTimeout})`,
    };
  }
  return fetch(tcUrl.href, {
    method: 'GET',
    headers: {
      'Content-Type': 'application/json',
    },
  })
    .then(async res => res.json())
    .catch(err => {
      return {
        models: {},
        error: `Error from topic-classification server: ${err.toString()}`,
      };
    });
}

export async function checkModelReady(arg: { model: string }) {
  const tcUrl = new URL('models', tcServer);
  tcUrl.searchParams.set('model', arg.model);
  const msTimeout = 1000;
  if (!(await isReachable(tcUrl.href, { timeout: msTimeout }))) {
    // TODO: move this backend check to server start-up time, maybe
    return { models: {}, error: `Topic Classification server is unreachable (${msTimeout})` };
  }
  return fetch(tcUrl.href, {
    method: 'GET',
    headers: {
      'Content-Type': 'application/json',
    },
  })
    .then(async res => res.json())
    .catch(err => {
      return {
        models: '',
        error: `Error from topic-classification server: ${err.toString()}`,
      };
    });
}

export async function processDocument(arg: { seq: string; model: string }) {
  const url = new URL(tcServer);
  url.searchParams.set('model', arg.model);
  const res = await fetch(url.pathname, {
    method: 'POST',
    body: JSON.stringify(arg.seq),
    headers: {
      'Content-Type': 'application/json',
    },
  });
  const body = await res.json();
  return body;
}
