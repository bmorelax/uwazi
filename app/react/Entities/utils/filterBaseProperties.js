export default {
  filterBaseProperties: (data) => {
    const properties = ['_id', 'language', 'metadata', 'sharedId', 'template', 'title', 'icon', 'type'];
    return Object.assign({}, ...properties.map(p => ({ [p]: data[p] })));
  }
};
