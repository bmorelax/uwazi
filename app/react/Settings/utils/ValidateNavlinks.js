export default function (links) {
  const validator = {
    '': {}
  };

  links.forEach((link, index) => {
    validator[''][`links.${index}.title.required`] = (form) => {
      if (!form.links[index]) {
        return true;
      }
      const { title } = form.links[index];
      return Boolean(title && String(title).trim().length);
    };
  });

  return validator;
}
