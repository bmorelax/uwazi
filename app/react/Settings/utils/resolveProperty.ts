/**
 * A given template property may refer to an existing thesaurus to provide
 * multi-select values. This function resolves a template property name that
 * refers to a particular thesaurus.
 *
 * @format
 */

export function resolveTemplateProp(thesaurus: any, templates: Array<any>) {
  let matchingProp;
  for (let i = 0; i < templates.length; i += 1) {
    const template = templates[i];
    const matchProp = template.properties.find((prop: any) => prop.content === thesaurus._id);
    if (matchProp !== undefined) {
      matchingProp = matchProp;
      // TODO: Consider supporting multiple fields referring to the same thesaurus.
      break;
    }
  }
  return matchingProp;
}
