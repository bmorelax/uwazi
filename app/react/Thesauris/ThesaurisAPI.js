import api from 'app/utils/api';

export default {
  get(requestParams) {
    const url = 'thesauris';
    return api.get(url, requestParams)
    .then(response => response.json.rows);
  },

  getDictionaries(requestParams) {
    const url = 'dictionaries';
    return api.get(url, requestParams)
    .then(response => response.json.rows);
  },

  save(requestParams) {
    return api.post('thesauris', requestParams)
    .then(response => response.json);
  },

  delete(requestParams) {
    return api.delete('thesauris', requestParams)
    .then(response => response.json);
  }
};
