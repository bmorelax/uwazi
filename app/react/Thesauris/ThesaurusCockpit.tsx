/** @format */
import RouteHandler from 'app/App/RouteHandler';
import { actions } from 'app/BasicReducer';
import { I18NLink, t } from 'app/I18N';
import api from 'app/Search/SearchAPI';
import { resolveTemplateProp } from 'app/Settings/utils/resolveProperty';
import { getSuggestionsQuery } from 'app/Settings/utils/suggestions';
import TemplatesAPI from 'app/Templates/TemplatesAPI';
import ThesaurisAPI from 'app/Thesauris/ThesaurisAPI';
import React from 'react';
import { connect } from 'react-redux';
import { createSelector } from 'reselect';
import { Icon } from 'UI';

/** Model is the type used for holding information about a classifier model. */
interface ClassifierModel {
  name: string;
  preferred: string;
  bert: string;
  sample: number;
  topics: {
    [key: string]: {
      name: string;
      quality: number;
      samples: number;
    };
  };
}

interface ThesaurusTopic {
  id: string;
  label: string;
  suggestions: number;
}

export class ThesaurusCockpitBase extends RouteHandler {
  static genIcons(label: string, actual: number, possible: number) {
    const icons = [];
    for (let i = 0; i < possible; i += 1) {
      let iconClass: any;
      if (i < actual) {
        iconClass = 'circle';
      } else {
        iconClass = ['far', 'circle'];
      }
      icons.push(<Icon key={`${label}_${i}`} icon={iconClass} />);
    }
    return icons;
  }

  static qualityIcon(label: string, val: number) {
    switch (true) {
      case val > 0.85:
        return (
          <div key={label} className="quality-icon quality-high">
            {ThesaurusCockpitBase.genIcons(label, 3, 3)}
          </div>
        );
      case val > 0.7:
        return (
          <div key={label} className="quality-icon quality-med">
            {ThesaurusCockpitBase.genIcons(label, 2, 3)}
          </div>
        );
      default:
        return (
          <div key={label} className="quality-icon quality-low">
            {ThesaurusCockpitBase.genIcons(label, 1, 3)}
          </div>
        );
    }
  }

  static topicNode(
    topic: ThesaurusTopic,
    modelInfo: ClassifierModel,
    propName: string | undefined
  ) {
    if (modelInfo === undefined) {
      return null;
    }
    const { label, id, suggestions } = topic;
    const { quality = 0 } = (modelInfo.topics || {})[label] || {};

    return (
      <tr key={label}>
        <th scope="row">{label}</th>
        <td title="quality-icons">{this.qualityIcon(label, quality)}</td>
        <td title="suggestions-count">{suggestions || null}</td>
        <td title="review-button">
          {suggestions > 0 && propName ? (
            <I18NLink
              to={`/review?q=(filters:(_${propName}:(values:!('${id}')),${propName}:(values:!(missing))))&includeUnpublished=1`}
              className="btn btn-default btn-xs"
            >
              <Icon icon="gavel" />
              &nbsp;
              <span title="review-button-title">{t('system', 'Review suggestions')}</span>
            </I18NLink>
          ) : null}
        </td>
      </tr>
    );
  }

  topicNodes() {
    const { values: topics, name, property } = this.props.thesaurus; // {name: Themes; values: [{label: Education}, ...]}

    if (!topics) {
      return null;
    }

    const model = this.props.models.find((modelInfo: ClassifierModel) => modelInfo.name === name);
    const propName: string | undefined = (this.props.thesaurus.property || {}).name;
    const { suggestions } = this.props;
    const thesaurusSuggestions: any = [];
    suggestions.forEach((templateResult: any) => {
      if (
        templateResult.aggregations !== undefined &&
        templateResult.aggregations.all.hasOwnProperty(`_${property.name}`)
      ) {
        const { buckets } = templateResult.aggregations.all[`_${property.name}`];
        buckets.forEach((b: any) => thesaurusSuggestions.push(b));
      }
    });
    const dedupedSuggestions = thesaurusSuggestions.reduce((acc: any, obj: any) => {
      const { key } = obj;
      if (!acc[key]) {
        acc[key] = obj.filtered.doc_count;
      } else {
        acc[key] += obj.filtered.doc_count;
      }
      return acc;
    }, {});

    topics.sort(
      // Sort in order of descending number of documents with suggestions
      // TODO: Make sort order configurable, or even better, dynamic
      (topic1: ThesaurusTopic, topic2: ThesaurusTopic) =>
        (dedupedSuggestions[topic2.id] || 0) - (dedupedSuggestions[topic1.id] || 0)
    );

    return topics.map((topic: ThesaurusTopic) => {
      const topicWithSuggestions = { ...topic, suggestions: dedupedSuggestions[topic.id] || 0 };
      return ThesaurusCockpitBase.topicNode(topicWithSuggestions, model, propName);
    });
  }

  static async requestState(requestParams: any) {
    // Thesauri should always have a length of 1, because a specific thesaurus ID is passed in the requestParams.
    const [thesauri, templates] = await Promise.all([
      ThesaurisAPI.getThesauri(requestParams),
      TemplatesAPI.get(requestParams.onlyHeaders()),
    ]);
    const thesaurus = thesauri[0];

    // Fetch models associated with known thesauri.
    const modelParams = requestParams.onlyHeaders().set({ model: thesaurus.name });
    const model: ClassifierModel = await ThesaurisAPI.getModelStatus(modelParams);

    // Get aggregate document count of documents with suggestions on this thesaurus
    const assocProp = resolveTemplateProp(thesaurus, templates);
    const allDocsWithSuggestions = await Promise.all(
      templates.map((template: { _id: string }) => {
        const reqParams = requestParams.set(getSuggestionsQuery(assocProp, template._id));
        return api.search(reqParams);
      })
    );
    thesaurus.property = assocProp;

    return [
      actions.set('thesauri/thesaurus', thesaurus),
      actions.set('thesauri/models', [model]),
      actions.set('thesauri/suggestions', allDocsWithSuggestions),
    ];
  }

  componentWillUnmount() {
    this.emptyState();
  }

  emptyState() {
    this.context.store.dispatch(actions.unset('thesauri/models'));
    this.context.store.dispatch(actions.unset('thesauri/thesaurus'));
    this.context.store.dispatch(actions.unset('thesauri/suggestions'));
  }

  suggestionsButton() {
    if (this.props.thesaurus === undefined || this.props.thesaurus.property === undefined) {
      return null;
    }
    const thesaurusPropertyRefName = this.props.thesaurus.property.name;

    return (
      <I18NLink
        to={
          `/uploads/?q=(filters:(_${thesaurusPropertyRefName}:(values:!(any)),${thesaurusPropertyRefName}:(values:!(any))),` +
          'limit:100,order:desc,sort:creationDate)&view=nosearch'
        }
        className="btn btn-primary btn-xs"
      >
        <Icon icon="search" />
        &nbsp;
        <span>{t('System', 'Review & Publish')}</span>
      </I18NLink>
    );
  }

  render() {
    const { name } = this.props.thesaurus; // {name: Themes; values: [{label: Education}, ...]}
    return (
      <div className="panel panel-default">
        <div className="panel-heading">
          {t('System', `Thesauri > ${name}`)}
          {this.suggestionsButton()}
        </div>
        <div className="cockpit">
          <table>
            <thead>
              <tr>
                <th scope="col" />
                <th scope="col">{t('System', 'Confidence')}</th>
                <th scope="col">{t('System', 'Suggestions')}</th>
                <th scope="col" />
              </tr>
            </thead>
            <tbody>{this.topicNodes()}</tbody>
          </table>
        </div>
        <div className="settings-footer">
          <I18NLink to="/settings/dictionaries" className="btn btn-default">
            <Icon icon="arrow-left" />
            <span className="btn-label">{t('System', 'Back')}</span>
          </I18NLink>
        </div>
      </div>
    );
  }
}

const selectModels = createSelector(
  (state: any) => state.thesauri.models,
  models => models.toJS()
);

const selectThesaurus = createSelector(
  (state: any) => state.thesauri.thesaurus,
  thesaurus => thesaurus.toJS()
);

const selectSuggestions = createSelector(
  (state: any) => state.thesauri.suggestions,
  suggestions => suggestions.toJS()
);

function mapStateToProps(state: any) {
  return {
    models: selectModels(state), // {name: ModelName; bert: bert123; sample: 53}
    thesaurus: selectThesaurus(state), // {name: Themes; values: [{label: Education, id: lkajsdf}, ...]}
    suggestions: selectSuggestions(state),
  };
}

export default connect(
  mapStateToProps,
  null
  //{ withRef: true }
)(ThesaurusCockpitBase);
