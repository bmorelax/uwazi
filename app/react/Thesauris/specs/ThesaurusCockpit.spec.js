/** @format */
import RouteHandler from 'app/App/RouteHandler';
import api from 'app/Search/SearchAPI';
import TemplatesAPI from 'app/Templates/TemplatesAPI';
import ThesaurisAPI from 'app/Thesauris/ThesaurisAPI';
import { RequestParams } from 'app/utils/RequestParams';
import { shallow } from 'enzyme';
import React from 'react';

import { ThesaurusCockpitBase } from '../ThesaurusCockpit';

const templates = [
  {
    _id: 'underscoreID',
    name: 'Paragraph',
    properties: [
      {
        name: 'thesaurus_name',
        label: 'ThesaurusName',
        content: 'thesaurusUnderscoreId1',
        type: 'multiselect',
      },
    ],
    default: true,
  },
  {
    _id: 'underscoreID2',
    name: 'Recommendation',
    properties: [
      {
        name: 'recommendation',
        label: 'Recommendation',
        content: 'thesaurusUnderscoreId2',
        type: 'multiselect',
      },
    ],
    default: false,
  },
];
const models = [
  {
    bert: 'testBert',
    completeness: 0,
    extraneous: 0,
    instances: ['timestamp'],
    name: 'ThesaurusName',
    preferred: 'timestamp',
    topics: {
      'Topic 1': {
        name: 'Topic 1',
        quality: 0.8,
        samples: 20,
      },
      'Topic 2': {
        name: 'Topic 2',
        quality: 0.92,
        samples: 25,
      },
      'Topic 3': {
        name: 'Topic 3',
        quality: 0.62,
        samples: 2,
      },
    },
  },
];
const thesauri = [
  {
    _id: 'thesaurusUnderscoreId1',
    name: 'ThesaurusName',
    values: [
      { _id: 'underscoreId1', label: 'Topic 1', id: 'id1' },
      { _id: 'underscoreId2', label: 'Topic 2', id: 'id2' },
      { _id: 'underscoreId3', label: 'Topic 3', id: 'id3' },
    ],
    enable_classification: true,
  },
  {
    _id: 'thesaurusUnderscoreId2',
    name: 'ThesaurusWithoutSuggestions',
    values: [{ _id: 'underscoreId1', label: 'Topic 1', id: 'id1' }],
    enable_classification: false,
  },
];
const suggestions = [
  {
    aggregations: { all: {} },
    rows: {},
    totalRows: 0,
  },
  {
    aggregations: {
      all: {
        doc_count: 65460,
        thesaurus_name: {
          buckets: [],
        },
        _thesaurus_name: {
          buckets: [
            {
              key: 'id1',
              doc_count: 8528,
              filtered: {
                doc_count: 1,
                meta: {},
              },
            },
            { key: 'id2', doc_count: 6923, filtered: {} },
          ],
        },
      },
    },
    rows: {
      sharedId: 'jo9dvda1ux',
      snippets: '',
      count: 0,
      fullText: [],
      metadata: [],
      _explanation: {},
      _id: 'docId1',
    },
    totalRows: 1,
  },
];

describe('ThesaurusCockpit', () => {
  describe('render', () => {
    let component;
    let props;
    let context;
    let dispatchCallsOrder = [];

    beforeEach(() => {
      const thesaurus = thesauri[0];
      // The render function has already mapped this property back to the thesaurus by now
      thesaurus.property = {
        content: 'content1',
        label: 'ThesaurusName',
        name: 'thesaurus_name',
      };
      props = { models, thesaurus: thesauri[0], suggestions };
      RouteHandler.renderedFromServer = true;
      dispatchCallsOrder = [];
      context = {
        subscribe: jest.fn(), // Find out why this needs to be here.
        confirm: jasmine.createSpy('confirm'),
        store: {
          getState: () => ({}),
          dispatch: jasmine.createSpy('dispatch').and.callFake(action => {
            dispatchCallsOrder.push(action.type);
          }),
        },
      };
    });

    const render = () => {
      component = shallow(<ThesaurusCockpitBase {...props} />, { context });
    };

    it('should render a ThesaurusCockpit', () => {
      render();
      expect(component).toMatchSnapshot();
    });

    it('should find the cockpit table and verify names, values and quality icons', () => {
      render();
      expect(component.find('.cockpit').length).toBe(1);
      expect(component.find({ scope: 'row' }).length).toBe(3);
      /* We expect 5 data cells -- three with quality icons and 1 each of
    suggestion counts and a review button */
      expect(component.find('td').children().length).toBe(5);
      expect(component.find({ title: 'review-button-title' }).length).toBe(1);
      expect(component.contains(<td title="suggestions-count">{1}</td>)).toEqual(true);
    });
  });

  describe('requestState', () => {
    beforeEach(() => {
      spyOn(ThesaurisAPI, 'getThesauri').and.returnValue(Promise.resolve(thesauri));
      spyOn(ThesaurisAPI, 'getModelStatus').and.returnValue(Promise.resolve(models));
      spyOn(TemplatesAPI, 'get').and.returnValue(Promise.resolve(templates));
      spyOn(api, 'search').and.returnValue(Promise.resolve(suggestions));
    });

    it('should get the thesaurus, classification model and suggestion counts as react actions', async () => {
      const actions = await ThesaurusCockpitBase.requestState(new RequestParams());
      expect(actions).toMatchSnapshot();
    });
  });
});
