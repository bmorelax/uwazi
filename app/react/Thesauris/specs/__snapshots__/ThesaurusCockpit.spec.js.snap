// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThesaurusCockpit render should render a ThesaurusCockpit 1`] = `
<div
  className="panel panel-default"
>
  <div
    className="panel-heading"
  >
    Thesauri &gt; ThesaurusName
    <Connect(I18NLink)
      className="btn btn-primary btn-xs"
      to="/uploads/?q=(filters:(_thesaurus_name:(values:!(any)),thesaurus_name:(values:!(any))),limit:100,order:desc,sort:creationDate)&view=nosearch"
    >
      <Connect(Icon)
        icon="search"
      />
       
      <span>
        Review & Publish
      </span>
    </Connect(I18NLink)>
  </div>
  <div
    className="cockpit"
  >
    <table>
      <thead>
        <tr>
          <th
            scope="col"
          />
          <th
            scope="col"
          >
            Confidence
          </th>
          <th
            scope="col"
          >
            Suggestions
          </th>
          <th
            scope="col"
          />
        </tr>
      </thead>
      <tbody>
        <tr
          key="Topic 1"
        >
          <th
            scope="row"
          >
            Topic 1
          </th>
          <td
            title="quality-icons"
          >
            <div
              className="quality-icon quality-med"
              key="Topic 1"
            >
              <Connect(Icon)
                icon="circle"
                key="Topic 1_0"
              />
              <Connect(Icon)
                icon="circle"
                key="Topic 1_1"
              />
              <Connect(Icon)
                icon={
                  Array [
                    "far",
                    "circle",
                  ]
                }
                key="Topic 1_2"
              />
            </div>
          </td>
          <td
            title="suggestions-count"
          >
            1
          </td>
          <td
            title="review-button"
          >
            <Connect(I18NLink)
              className="btn btn-default btn-xs"
              to="/review?q=(filters:(_thesaurus_name:(values:!('id1')),thesaurus_name:(values:!(missing))))&includeUnpublished=1"
            >
              <Connect(Icon)
                icon="gavel"
              />
               
              <span
                title="review-button-title"
              >
                Review suggestions
              </span>
            </Connect(I18NLink)>
          </td>
        </tr>
        <tr
          key="Topic 2"
        >
          <th
            scope="row"
          >
            Topic 2
          </th>
          <td
            title="quality-icons"
          >
            <div
              className="quality-icon quality-high"
              key="Topic 2"
            >
              <Connect(Icon)
                icon="circle"
                key="Topic 2_0"
              />
              <Connect(Icon)
                icon="circle"
                key="Topic 2_1"
              />
              <Connect(Icon)
                icon="circle"
                key="Topic 2_2"
              />
            </div>
          </td>
          <td
            title="suggestions-count"
          />
          <td
            title="review-button"
          />
        </tr>
        <tr
          key="Topic 3"
        >
          <th
            scope="row"
          >
            Topic 3
          </th>
          <td
            title="quality-icons"
          >
            <div
              className="quality-icon quality-low"
              key="Topic 3"
            >
              <Connect(Icon)
                icon="circle"
                key="Topic 3_0"
              />
              <Connect(Icon)
                icon={
                  Array [
                    "far",
                    "circle",
                  ]
                }
                key="Topic 3_1"
              />
              <Connect(Icon)
                icon={
                  Array [
                    "far",
                    "circle",
                  ]
                }
                key="Topic 3_2"
              />
            </div>
          </td>
          <td
            title="suggestions-count"
          />
          <td
            title="review-button"
          />
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="settings-footer"
  >
    <Connect(I18NLink)
      className="btn btn-default"
      to="/settings/dictionaries"
    >
      <Connect(Icon)
        icon="arrow-left"
      />
      <span
        className="btn-label"
      >
        Back
      </span>
    </Connect(I18NLink)>
  </div>
</div>
`;

exports[`ThesaurusCockpit requestState should get the thesaurus, classification model and suggestion counts as react actions 1`] = `
Array [
  Object {
    "type": "thesauri/thesaurus/SET",
    "value": Object {
      "_id": "thesaurusUnderscoreId1",
      "enable_classification": true,
      "name": "ThesaurusName",
      "property": Object {
        "content": "thesaurusUnderscoreId1",
        "label": "ThesaurusName",
        "name": "thesaurus_name",
        "type": "multiselect",
      },
      "values": Array [
        Object {
          "_id": "underscoreId1",
          "id": "id1",
          "label": "Topic 1",
        },
        Object {
          "_id": "underscoreId2",
          "id": "id2",
          "label": "Topic 2",
        },
        Object {
          "_id": "underscoreId3",
          "id": "id3",
          "label": "Topic 3",
        },
      ],
    },
  },
  Object {
    "type": "thesauri/models/SET",
    "value": Array [
      Array [
        Object {
          "bert": "testBert",
          "completeness": 0,
          "extraneous": 0,
          "instances": Array [
            "timestamp",
          ],
          "name": "ThesaurusName",
          "preferred": "timestamp",
          "topics": Object {
            "Topic 1": Object {
              "name": "Topic 1",
              "quality": 0.8,
              "samples": 20,
            },
            "Topic 2": Object {
              "name": "Topic 2",
              "quality": 0.92,
              "samples": 25,
            },
            "Topic 3": Object {
              "name": "Topic 3",
              "quality": 0.62,
              "samples": 2,
            },
          },
        },
      ],
    ],
  },
  Object {
    "type": "thesauri/suggestions/SET",
    "value": Array [
      Array [
        Object {
          "aggregations": Object {
            "all": Object {},
          },
          "rows": Object {},
          "totalRows": 0,
        },
        Object {
          "aggregations": Object {
            "all": Object {
              "_thesaurus_name": Object {
                "buckets": Array [
                  Object {
                    "doc_count": 8528,
                    "filtered": Object {
                      "doc_count": 1,
                      "meta": Object {},
                    },
                    "key": "id1",
                  },
                  Object {
                    "doc_count": 6923,
                    "filtered": Object {},
                    "key": "id2",
                  },
                ],
              },
              "doc_count": 65460,
              "thesaurus_name": Object {
                "buckets": Array [],
              },
            },
          },
          "rows": Object {
            "_explanation": Object {},
            "_id": "docId1",
            "count": 0,
            "fullText": Array [],
            "metadata": Array [],
            "sharedId": "jo9dvda1ux",
            "snippets": "",
          },
          "totalRows": 1,
        },
      ],
      Array [
        Object {
          "aggregations": Object {
            "all": Object {},
          },
          "rows": Object {},
          "totalRows": 0,
        },
        Object {
          "aggregations": Object {
            "all": Object {
              "_thesaurus_name": Object {
                "buckets": Array [
                  Object {
                    "doc_count": 8528,
                    "filtered": Object {
                      "doc_count": 1,
                      "meta": Object {},
                    },
                    "key": "id1",
                  },
                  Object {
                    "doc_count": 6923,
                    "filtered": Object {},
                    "key": "id2",
                  },
                ],
              },
              "doc_count": 65460,
              "thesaurus_name": Object {
                "buckets": Array [],
              },
            },
          },
          "rows": Object {
            "_explanation": Object {},
            "_id": "docId1",
            "count": 0,
            "fullText": Array [],
            "metadata": Array [],
            "sharedId": "jo9dvda1ux",
            "snippets": "",
          },
          "totalRows": 1,
        },
      ],
    ],
  },
]
`;
